### ISSUE TITLE
Fluent Speech Commands WER wrong since `eos` is counted as token

### ISSUE THREAD

**Issue by tomaz-suller** [speechbrain/speechbrain#2863]
### Describe the bug

All predictions from both the pre-trained model and those trained locally have their WER computed incorrectly since the computation takes the `eos` token, represented as ` ⁇ `, into account.

An example from `wer_test.txt` generated by the recipe is

```
4580, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;      =      ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ;   ⁇   ;      
```

### Expected behaviour

`eos` should be removed before computing the metric.

In the example provided above:

```
4580, %WER 0.0 [ 0 / 6, 0 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"}
    =      ;      =      ;     =     ;    =    ;      =      ;      =     
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"}
```

### To Reproduce

Execute `fluent-speech-commands` recipe following the in the `README.md` inside the corresponding directory.

### Environment Details

* Python 3.10 
* PyTorch 1.13.1 compiled from source with CUDA support
* torchaudio 0.13.1
* CUDA 11.6
* SpeechBrain 1.0.2

### Relevant Log Output

```shell
%WER 32.82 [ 7604 / 23166, 7586 ins, 0 del, 18 sub ]
%SER 100.00 [ 3793 / 3793 ]
Scored 3793 sentences, 0 not present in hyp.
================================================================================
ALIGNMENTS

Format:
<utterance-id>, WER DETAILS
<eps> ; reference  ; on ; the ; first ;  line
  I   ;     S      ; =  ;  =  ;   S   ;   D  
 and  ; hypothesis ; on ; the ; third ; <eps>
================================================================================
4580, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;      =      ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ;   ⁇   ;      
================================================================================
4160, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "deactivate"| ; "object": ; "music"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;       =       ;     =     ;    =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "deactivate"| ; "object": ; "music"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
4597, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "kitchen"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;     =      ;   I   ;   I  
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "kitchen"} ;   ⁇   ;      
================================================================================
4158, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "deactivate"| ; "object": ; "music"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;       =       ;     =     ;    =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "deactivate"| ; "object": ; "music"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
3388, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;    =    ;   I   ;   I  
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
3495, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "bedroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;     =      ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "bedroom"} ;   ⁇   ;      
================================================================================
5791, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "activate"| ; "object": ; "lights"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "activate"| ; "object": ; "lights"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
5451, %WER 28.57 [ 2 / 7, 2 ins, 0 del, 0 sub ]
{"action:" ; "change ; language"| ; "object": ; "none"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;    =    ;     =      ;     =     ;    =    ;      =      ;    =    ;   I   ;   I  
{"action:" ; "change ; language"| ; "object": ; "none"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
4974, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
6204, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;    =    ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
6179, %WER 28.57 [ 2 / 7, 2 ins, 0 del, 0 sub ]
{"action:" ; "change ; language"| ; "object": ; "Korean"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;    =    ;     =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "change ; language"| ; "object": ; "Korean"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
4427, %WER 28.57 [ 2 / 7, 2 ins, 0 del, 0 sub ]
{"action:" ; "change ; language"| ; "object": ; "Korean"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;    =    ;     =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "change ; language"| ; "object": ; "Korean"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
6138, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "bring"| ; "object": ; "newspaper"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;    =     ;     =     ;      =       ;      =      ;    =    ;   I   ;   I  
{"action:" ; "bring"| ; "object": ; "newspaper"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
5792, %WER 50.00 [ 3 / 6, 2 ins, 0 del, 1 sub ]
{"action:" ;  "activate"|  ; "object": ; "lights"| ; "location": ; "kitchen"} ; <eps> ; <eps>
    =      ;       S       ;     =     ;     =     ;      =      ;     =      ;   I   ;   I  
{"action:" ; "deactivate"| ; "object": ; "lights"| ; "location": ; "kitchen"} ;   ⁇   ;      
================================================================================
6342, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "kitchen"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;     =      ;   I   ;   I  
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "kitchen"} ;   ⁇   ;      
================================================================================
6091, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;      =      ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ;   ⁇   ;      
================================================================================
5552, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
6403, %WER 28.57 [ 2 / 7, 2 ins, 0 del, 0 sub ]
{"action:" ; "change ; language"| ; "object": ; "German"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;    =    ;     =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "change ; language"| ; "object": ; "German"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
5899, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "activate"| ; "object": ; "lights"| ; "location": ; "kitchen"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;     =     ;      =      ;     =      ;   I   ;   I  
{"action:" ; "activate"| ; "object": ; "lights"| ; "location": ; "kitchen"} ;   ⁇   ;      
================================================================================
4615, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "washroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;      =      ;   I   ;   I  
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "washroom"} ;   ⁇   ;      
================================================================================
5133, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
4620, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "washroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;      =      ;   I   ;   I  
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "washroom"} ;   ⁇   ;      
================================================================================
5453, %WER 28.57 [ 2 / 7, 2 ins, 0 del, 0 sub ]
{"action:" ; "change ; language"| ; "object": ; "Chinese"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;    =    ;     =      ;     =     ;     =      ;      =      ;    =    ;   I   ;   I  
{"action:" ; "change ; language"| ; "object": ; "Chinese"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
4269, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "deactivate"| ; "object": ; "lights"| ; "location": ; "bedroom"} ; <eps> ; <eps>
    =      ;       =       ;     =     ;     =     ;      =      ;     =      ;   I   ;   I  
{"action:" ; "deactivate"| ; "object": ; "lights"| ; "location": ; "bedroom"} ;   ⁇   ;      
================================================================================
4988, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
5076, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "deactivate"| ; "object": ; "lamp"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;       =       ;     =     ;    =    ;      =      ;    =    ;   I   ;   I  
{"action:" ; "deactivate"| ; "object": ; "lamp"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
3653, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "activate"| ; "object": ; "lights"| ; "location": ; "bedroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;     =     ;      =      ;     =      ;   I   ;   I  
{"action:" ; "activate"| ; "object": ; "lights"| ; "location": ; "bedroom"} ;   ⁇   ;      
================================================================================
6215, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "activate"| ; "object": ; "music"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "activate"| ; "object": ; "music"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
4081, %WER 28.57 [ 2 / 7, 2 ins, 0 del, 0 sub ]
{"action:" ; "change ; language"| ; "object": ; "Korean"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;    =    ;     =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "change ; language"| ; "object": ; "Korean"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
6337, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;    =    ;   I   ;   I  
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
3854, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "deactivate"| ; "object": ; "lights"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;       =       ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "deactivate"| ; "object": ; "lights"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
5025, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;      =      ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ;   ⁇   ;      
================================================================================
4178, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;     =     ;      =      ;    =    ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "volume"| ; "location": ; "none"} ;   ⁇   ;      
================================================================================
5174, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "bedroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;     =      ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "bedroom"} ;   ⁇   ;      
================================================================================
4605, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "bedroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;     =      ;   I   ;   I  
{"action:" ; "decrease"| ; "object": ; "heat"| ; "location": ; "bedroom"} ;   ⁇   ;      
================================================================================
6568, %WER 33.33 [ 2 / 6, 2 ins, 0 del, 0 sub ]
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ; <eps> ; <eps>
    =      ;      =      ;     =     ;    =    ;      =      ;      =      ;   I   ;   I  
{"action:" ; "increase"| ; "object": ; "heat"| ; "location": ; "washroom"} ;   ⁇   ;
```

**Comment by Adel-Moumen** [comment]
Hey @tomaz-suller, I will try to get back to you ASAP. I assume that this issue should be reproducible with a different dataset like LibriSpeech.

**Comment by tomaz-suller** [comment]
Thanks! I assume so but I've only used FSC so far, and even then I've implemented a workaround on my code to compute metrics correctly despite that
